From inaky@linux.intel.com  Tue Jan 20 17:09:31 2009
From: Inaky Perez-Gonzalez <inaky@linux.intel.com>
Date: Tue, 20 Jan 2009 12:17:28 -0800
Subject: debugfs: introduce stub for debugfs_create_size_t() when DEBUG_FS=n
To: Toralf Förster <toralf.foerster@gmx.de>, Greg KH <greg@kroah.com>
Cc: wimax@linuxwimax.org, linux-kernel@vger.kernel.org
Message-ID: <1232482648-2663-1-git-send-email-inaky@linux.intel.com>

From: Inaky Perez-Gonzalez <inaky@linux.intel.com>

Toralf Förster <toralf.foerster@gmx.de> reported a build failure in
the WiMAX stack when CONFIG_DEBUG_FS=n

http://linuxwimax.org/pipermail/wimax/2009-January/000449.html

This is due to debugfs_create_size_t() missing an stub that returns
-ENODEV when the DEBUGFS subsystem is not configured in (like the rest
of the debugfs API).

This patch adds said stub.

Reported-by: Toralf Förster <toralf.foerster@gmx.de>
Signed-off-by: Inaky Perez-Gonzalez <inaky@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/debugfs.h |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/include/linux/debugfs.h
+++ b/include/linux/debugfs.h
@@ -162,6 +162,14 @@ static inline struct dentry *debugfs_cre
 	return ERR_PTR(-ENODEV);
 }
 
+static inline struct dentry *debugfs_create_size_t(const char *name,
+						   mode_t mode,
+						   struct dentry *parent,
+						   size_t *value)
+{
+	return ERR_PTR(-ENODEV);
+}
+
 static inline struct dentry *debugfs_create_bool(const char *name, mode_t mode,
 						 struct dentry *parent,
 						 u32 *value)
